### PR TITLE
view commits

### PR DESCRIPTION
**Pull request by Wubuntu88**
Initial commit of the checkers GUI for model testing.

### CODE REVIEW

File: src/view/CheckersPane.java [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+		positionsOfImages.put(positionAsString, selectedImageView);
+		String piece = typeOfpieceAtPosition.get(keyToDelete);
+		typeOfpieceAtPosition.remove(keyToDelete);
+		typeOfpieceAtPosition.put(positionAsString, piece);
+		
+		
+	}
+	
+	
+	Boolean pieceHasFinishedMoving = false;
```
Comment (csperlich) [inline src/view/CheckersPane.java]: I don't think the variable pieceHasFinishedMoving is used anywhere.

File: src/view/CheckersPane.java [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+        int row = (int) y / (int)squareHeight;
+        return row;
+    }
+	
+}
+
+
+
+
+
```
Comment (csperlich) [inline src/view/CheckersPane.java]: I believe this file needs a re-indentation. (auto-format)

File: src/controller/CheckersController.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    public void computerFinishedMove() {
+        System.out.println("computer finished");
+        // TODO Auto-generated method stub
+        
+    }
+}
+
+
+
+
```
Comment (csperlich) [inline src/controller/CheckersController.java]: I think this file might could use a re-indentation as well.

File: src/controller/CheckersController.java [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    @Override
+    public void humanFinishedMove(ArrayList<String> moves) {
+        System.out.println("human finished");
+        
+        //let computer calculate move
+        //execute the move method of the view to reflect this
+        //have a handler in the view for the computer finished moving.
+        //when the computer's move has finished
+    }
+    @Override
```
Comment (csperlich) [inline src/controller/CheckersController.java]: This file could use a re-indentation.

File: src/view/CheckersPane.java [hunk 5]
```diff
@@ -0,0 +1,491 @@
...
+	    drawBoard(gc);
+	    
+	    Image zerlingImage = new Image("zergling.jpeg");
+        Image stalkerImage = new Image("stalker.jpeg");
+	    
+	    List<Square> squares = board.getGameState();
+	    for(Square square : squares){
+            if(square.isOccupied()){
+                PieceInterface piece = square.getOccupyingPiece();
+                //I use row-1 because row in model is 1, same row in view is 0
```
Comment (csperlich) [inline src/view/CheckersPane.java]: This file could use a re-indentation as well.

**Comment by csperlich** [comment]
I think this looks pretty good man.

**Comment by stefprez** [comment]
Yeah, there's a lot of stuff on the GUI side that I'm just not very familiar with, so I say we get it merged in, that way I can play with it in context. The only thing I think of is doing a bit of refactoring and cleanup. For example, I think the ComputerFinishedMove and HumanFinishedMove interfaces are unnecessary. But that's all minor stuff. I say let's get it all together right now, and cleanup can occur later.

**Comment by csperlich** [comment]
Agreed, I am going to merge it.

**Pull request merged**